### ISSUE TITLE
Processes stuck in SUSPENDED after releasing lock when using parallel

### ISSUE THREAD

**Issue by ppalucha** [walmartlabs/concord#367]
We have processes stuck in SUSPENDED state when waiting lock is released. Having flows (v2):
```
flows:
  lockParent:
    - parallel:
        - call: lockChild1
        - call: lockChild2
  lockChild1:
    - task: lock
      in:
        name: Lock1
        scope: PROJECT
    - ${sleep.ms(30000)}
    - task: unlock
      in:
        name: Lock1
        scope: Project
  lockChild2:
    - task: lock
      in:
        name: Lock2
        scope: PROJECT
    - ${sleep.ms(30000)}
    - task: unlock
      in:
        name: Lock2
        scope: Project
```
When running lockParent flow twice in parallel (within 30 seconds):
- First process acquires both locks
- Second process suspends on both locks
- First process releases locks
- Second process acquires first lock, but never never acquires second lock
As a result second lock is stuck in SUSPENDED state, process_locks table in DB is empty.

**Comment by ppalucha** [comment]
It seems the problem is that wait_condition on process can hold information about single lock only. So, if locks are acquired by 2 threads, information about one of them is lost.

**Comment by brig** [comment]
yep, process can hold only one lock.

**Comment by ibodrov** [comment]
The fix is merged into master, should be released in 1.84.0 this week.
